### PR TITLE
move scripts into subpackage so that they can be packaged with `entry_points.console_scripts`

### LINKED ISSUE #4105: Convert scripts to entrypoints
We should review the status of these scripts and consolidate the useful ones back into their parent pipeline step. I propose doing it like this `jwst/[step_name]/cli.py` or this  `jwst/[step_name]/cli/name_here.py` so it will be possible to generate `entry_points` in `setup.py`  instead of curating them all by hand.

## Scripts
```bash 
$ grep -E -l -r 'argv\[.*\]|(opt|arg)parse' *                                                                                                                                                                          
jwst/associations/lib/lv2_v1_to_v2.py
jwst/associations/main.py
jwst/associations/asn_from_list.py
jwst/datamodels/schema_editor.py
jwst/exp_to_source/main.py
jwst/csv_tools/csvconvert.py
jwst/ramp_fitting/mc_3d.py
jwst/ramp_fitting/mc_2d.py
jwst/ramp_fitting/create_cube.py
jwst/ramp_fitting/rotate_dataset.py
jwst/ramp_fitting/compare_cr_navg_files.py
jwst/ramp_fitting/compare_cr_files.py
jwst/ramp_fitting/compare_crs.py
jwst/stpipe/cmdline.py
scripts/strun
scripts/utc_to_bary.py
scripts/jump
scripts/cube_build
scripts/create_data
scripts/straylight
scripts/set_velocity_aberration.py
scripts/assign_wcs
scripts/outlier_detection
scripts/stspec
scripts/linearitycorr
scripts/asn_edit
scripts/dark_current
scripts/persistencecorr
scripts/resample
scripts/data_generate
scripts/collect_pipeline_cfgs
scripts/set_bary_helio_times.py
scripts/okify_regression_tests.py
scripts/rampfitcorr
scripts/ipc
scripts/move_wcs.py
scripts/flatfieldcorr
scripts/make_header
scripts/set_telescope_pointing.py
scripts/schemadoc
scripts/world_coords
scripts/photomcorr
scripts/coron
```

### LINKED ISSUE #7387: `test_velocity_aberration_script ` fails when using `pytest-xdist`
```shell
tox -e test-xdist -- jwst/lib/tests/test_velocity_aberration.py
```
```
.pkg: _optional_hooks> python /Users/zburnett/mambaforge/envs/jwst/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_editable> python /Users/zburnett/mambaforge/envs/jwst/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_wheel> python /Users/zburnett/mambaforge/envs/jwst/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: build_editable> python /Users/zburnett/mambaforge/envs/jwst/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
test-xdist: install_package> python -I -m pip install --force-reinstall --no-deps /Users/zburnett/projects/jwst/.tox/.tmp/package/5/jwst-1.8.3.dev82+g6eee470b8.d20221209-0.editable-cp311-cp311-macosx_10_9_x86_64.whl
test-xdist: commands_pre[0] .> pip freeze
asdf==2.14.2
asdf-astropy==0.3.0
asdf-coordinates-schemas==0.1.0
asdf-standard==1.0.3
asdf-transform-schemas==0.3.0
asdf-unit-schemas==0.1.0
asdf-wcs-schemas==0.1.1
astropy==5.1.1
attrs==22.1.0
BayesicFitting==3.0.1
certifi==2022.12.7
charset-normalizer==2.1.1
ci-watson==0.6.1
codecov==2.1.12
colorama==0.4.6
contourpy==1.0.6
coverage==6.5.0
crds==11.16.16
cycler==0.11.0
drizzle==1.13.6
execnet==1.9.0
filelock==3.8.2
fonttools==4.38.0
future==0.18.2
gwcs==0.18.2
idna==3.4
imageio==2.22.4
iniconfig==1.1.1
jmespath==1.0.1
jsonschema==4.17.3
jwst @ file:///Users/zburnett/projects/jwst/.tox/.tmp/package/5/jwst-1.8.3.dev82%2Bg6eee470b8.d20221209-0.editable-cp311-cp311-macosx_10_9_x86_64.whl
kiwisolver==1.4.4
lxml==4.9.1
matplotlib==3.6.2
networkx==2.8.8
numpy==1.23.5
opencv-python==4.6.0.66
packaging==22.0
Parsley==1.3
photutils==1.5.0
Pillow==9.3.0
pluggy==1.0.0
poppy==1.0.3
psutil==5.9.4
pyerfa==2.0.0.1
pyparsing==3.0.9
pyrsistent==0.19.2
pytest==7.2.0
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-openfiles==0.5.0
pytest-xdist==3.1.0
python-dateutil==2.8.2
PyWavelets==1.4.1
PyYAML==6.0
readchar==4.0.3
requests==2.28.1
requests-mock==1.10.0
ruff==0.0.171
scikit-image==0.19.3
scipy==1.9.3
semantic-version==2.10.0
six==1.16.0
spherical-geometry==1.2.23
stcal==1.2.2
stdatamodels==0.4.3
stpipe==0.4.2
stsci.image==2.3.5
stsci.imagestats==1.6.3
stsci.stimage==0.2.6
tifffile==2022.10.10
tweakwcs==0.8.0
urllib3==1.26.13
wiimatch==0.3.1
test-xdist: commands[0] .> pytest -n auto jwst/lib/tests/test_velocity_aberration.py
================================================== test session starts ===================================================
platform darwin -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0
cachedir: .tox/test-xdist/.pytest_cache
crds_context: jwst_1019.pmap
rootdir: /Users/zburnett/projects/jwst, configfile: setup.cfg
plugins: asdf-2.14.2, xdist-3.1.0, requests-mock-1.10.0, jwst-1.8.3.dev82+g6eee470b8.d20221209, doctestplus-0.12.1, cov-4.0.0, ci-watson-0.6.1, openfiles-0.5.0
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I
gw0 [3] / gw1 [3] / gw2 [3] / gw3 [3] / gw4 [3] / gw5 [3] / gw6 [3] / gw7 [3]

..2022-12-09 10:16:27,873 - stpipe - WARNING - /Users/zburnett/projects/jwst/jwst/conftest.py:90: PytestDeprecationWarning: The hookimpl pytest_configure uses old-style configuration options (marks or attributes).
2022-12-09 10:16:27,873 - stpipe - WARNING - Please use the pytest.hookimpl(trylast=True) decorator instead
2022-12-09 10:16:27,873 - stpipe - WARNING -  to configure the hooks.
2022-12-09 10:16:27,873 - stpipe - WARNING -  See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
2022-12-09 10:16:27,873 - stpipe - WARNING -   @pytest.mark.trylast
2022-12-09 10:16:27,873 - stpipe - WARNING - 
F                                                                                                                [100%]
======================================================== FAILURES ========================================================
____________________________________________ test_velocity_aberration_script _____________________________________________
[gw2] darwin -- Python 3.11.0 /Users/zburnett/projects/jwst/.tox/test-xdist/bin/python

tmpdir = local('/private/var/folders/bv/nnh6jr1553jcv2pxbt8hyw6c00053_/T/pytest-of-zburnett/pytest-6/popen-gw2/test_velocity_aberration_scrip0')

    def test_velocity_aberration_script(tmpdir):
        """Test the whole script on a FITS file"""
        path = str(tmpdir.join("velocity_aberration_tmpfile.fits"))
        hdulist = fits.HDUList([fits.PrimaryHDU(), fits.ImageHDU(name='SCI')])
        hdulist['SCI'].header['JWST_DX'] = GOOD_VELOCITY[0]
        hdulist['SCI'].header['JWST_DY'] = GOOD_VELOCITY[1]
        hdulist['SCI'].header['JWST_DZ'] = GOOD_VELOCITY[2]
        hdulist['SCI'].header['RA_REF'] = GOOD_POS[0]
        hdulist['SCI'].header['DEC_REF'] = GOOD_POS[1]
        hdulist.writeto(path)
        hdulist.close()
    
>       subprocess.check_call(["set_velocity_aberration.py", path])

jwst/lib/tests/test_velocity_aberration.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

popenargs = (['set_velocity_aberration.py', '/private/var/folders/bv/nnh6jr1553jcv2pxbt8hyw6c00053_/T/pytest-of-zburnett/pytest-6/popen-gw2/test_velocity_aberration_scrip0/velocity_aberration_tmpfile.fits'],)
kwargs = {}, retcode = 1
cmd = ['set_velocity_aberration.py', '/private/var/folders/bv/nnh6jr1553jcv2pxbt8hyw6c00053_/T/pytest-of-zburnett/pytest-6/popen-gw2/test_velocity_aberration_scrip0/velocity_aberration_tmpfile.fits']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['set_velocity_aberration.py', '/private/var/folders/bv/nnh6jr1553jcv2pxbt8hyw6c00053_/T/pytest-of-zburnett/pytest-6/popen-gw2/test_velocity_aberration_scrip0/velocity_aberration_tmpfile.fits']' returned non-zero exit status 1.

../../mambaforge/envs/jwst/lib/python3.11/subprocess.py:413: CalledProcessError
================================================ short test summary info =================================================
FAILED jwst/lib/tests/test_velocity_aberration.py::test_velocity_aberration_script - subprocess.CalledProcessError: Com...
============================================== 1 failed, 2 passed in 4.19s ===============================================
test-xdist: exit 1 (6.07 seconds) .> pytest -n auto jwst/lib/tests/test_velocity_aberration.py pid=9414
.pkg: _exit> python /Users/zburnett/mambaforge/envs/jwst/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  test-xdist: FAIL code 1 (15.87=setup[9.16]+cmd[0.64,6.07] seconds)
  evaluation failed :( (16.09 seconds)
```

### LINKED ISSUE #7388: `module not found` when installing `editable` in Tox 4.0
when using `package = editable` to fix https://github.com/spacetelescope/jwst/issues/7386, the following error occurs for the `oldestdeps` run when running the `minimum_deps` script in `commands_pre`:
https://github.com/spacetelescope/jwst/actions/runs/3658899060/jobs/6184258846
```
pkg_resources.DistributionNotFound: The 'jwst' distribution was not found and is required by the application
```

### PR DESCRIPTION
**Pull request by zacharyburnett**
Resolves [JP-1616](https://jira.stsci.edu/browse/JP-1616)

closes #7388, closes #4105, closes #7387

This PR addresses the issue raised in #7388, where a PEP517 build does not properly install scripts packaged using the legacy `setup(scripts=...)` method for standalone scripts; this changes them to be packaged with `entry_points.console_scripts` as modules. 

This change **should** be transparent to the user, but just in case I added `test_scripts.py` to test if scripts are accessible under the same name after installation.

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/scripts/world_coords.py
Comment (zacharyburnett) [inline jwst/scripts/world_coords.py:354]: I added these two `#noqa` comments because it appears that `slit()` and `i` are both undefined, and I couldn't find where they are supposed to come from.

File: jwst/scripts/world_coords.py
Comment (zacharyburnett) [inline jwst/scripts/world_coords.py:347]: what is `slit()`?

File: jwst/scripts/world_coords.py
Comment (zacharyburnett) [inline jwst/scripts/world_coords.py:354]: where does `i` come from?

**Review by zacharyburnett — COMMENTED** [review]
I added these two `#noqa` comments because it appears that `slit()` and `i` are both undefined, and I couldn't figure out where they are supposed to come from.

File: jwst/scripts/asn_edit.py
Comment (stscieisenhamer) [inline jwst/scripts/asn_edit.py:38]: Very happy about reformatting, except the changing of quotes. Is there a reason?  Working with RSS, the extra shift key is a problem. Is this a `black` thing?

File: setup.cfg
Comment (stscieisenhamer) [inline setup.cfg:131]: Though an appropriate change, one I agree with, the changes for both `set_telescope_pointing.py` and `set_velocity_aberration.py` to loose the `.py`, is a public API deprecation. Some smoother transition will be necessary. 

In particular, this (almost) cannot happen for the DMS Build 9.1; this will affect the rest of the automatic processing and it is a bit late in the development cycle to be foisting such a change on SDP.

**Review by stscieisenhamer — CHANGES_REQUESTED** [review]
Overall, yea!

However, one issue and one comment about the reformatting:

- At least two command line commands are changing, loosing the ".py". This is a public API deprecation and should have a smoother treatment. Further commentary in the code.
- Changes in quote style. Is there a reason? Asking for someone who needs to minimize their extra-key usage.

**Review by kmacdonald-stsci — APPROVED** [review]

File: setup.cfg
Comment (zacharyburnett) [reply setup.cfg:131]: That's a good catch; I can add two more passthrough entry points and add warnings to them about future deprecation.

**Comment by zacharyburnett** [comment]
I will also work on moving these entry points into their respective subpackages, as they are no longer required to be in the same directory together

**Comment by zacharyburnett** [comment]
> * Changes in quote style. Is there a reason? Asking for someone who needs to minimize their extra-key usage.

This was just my auto-formatter, which I've set to `black` for standardization. I can change those back to minimize the code changes.

File: setup.cfg
Comment (zacharyburnett) [reply setup.cfg:131]: done in 4bd4e96

File: jwst/scripts/asn_edit.py
Comment (zacharyburnett) [reply jwst/scripts/asn_edit.py]: yeah that was an artifact from `black`; I reverted the quote changes and rebased

File: jwst/scripts/set_telescope_pointing.py
Comment (zacharyburnett) [inline jwst/scripts/set_telescope_pointing.py:207]: added this function to warn for use of deprecated entry point

File: jwst/scripts/set_velocity_aberration.py
Comment (zacharyburnett) [inline jwst/scripts/set_velocity_aberration.py:48]: added this function to warn for use of deprecated entry point

File: setup.cfg
Comment (zacharyburnett) [inline setup.cfg:134]: added the existing entry points back, so if they are called they warn to use the new one

**Review by zacharyburnett — COMMENTED** [review]
@stscieisenhamer do you think this is sufficient?

File: jwst/scripts/set_telescope_pointing.py
Comment (stscieisenhamer) [reply jwst/scripts/set_telescope_pointing.py:207]: This should work fine. Thanks!

**Review by stscieisenhamer — COMMENTED** [review]
In general LGTM pending others' approvals.

During Coffee standup, there was some discussion on this whole PR. I've added Howard to the reviewers

**Comment by zacharyburnett** [comment]
this PR blocks #7384

**Review by stscieisenhamer — APPROVED** [review]

File: jwst/scripts/v1_calculate.py
Comment (jdavies-st) [inline jwst/scripts/v1_calculate.py:127]: I know this is merged already, but generally, if you are installing via console scripts entry points, you no longer need this in the script itself, especially if you are pointing to `main()` in the entry point.

**Pull request merged**
